Sources/Framework/Controllers/PinEntryViewController.swift

```diff
@@ -85,3 +90,11 @@ class PinEntryViewController: UIViewController, PinEntryControllerProviding, Ale
     }
 
 }
+
+private func performLogIn(with actionHandler: LoginActionHandling?, withPin pin: String) -> Promise<VoteContext> {
```

Comment:
Validation logic should be moved to `LoginActionHandler` removing need to unwrap login action handler promise